fe/fe-core/src/main/java/com/starrocks/connector/ConnectorMgr.java

```diff
@@ -29,29 +29,36 @@ private void init() {
         addConnectorFactory(new HiveConnectorFactory());
     }
 
-    public synchronized void addConnectorFactory(ConnectorFactory connectorFactory) {
+    public void addConnectorFactory(ConnectorFactory connectorFactory) {
         Preconditions.checkNotNull(connectorFactory, "connectorFactory is null");
         ConnectorFactory existingConnectorFactory = connectorFactories.putIfAbsent(
                 connectorFactory.name(), connectorFactory);
         Preconditions.checkArgument(existingConnectorFactory == null,
-                "Connector '$s' is already registered", connectorFactory.name());
+                "ConnectorFactory '$s' is already registered", connectorFactory.name());
     }
 
     public Connector createConnector(ConnectorContext context) throws DdlException {
         String catalogName = context.getCatalogName();
         String type = context.getType();
         ConnectorFactory connectorFactory = connectorFactories.get(type);
         Preconditions.checkNotNull(connectorFactory, "Cannot load %s connector factory", type);
-        Preconditions.checkState(!connectors.containsKey(catalogName), "Catalog '%s' already exists", catalogName);
+        Preconditions.checkState(!connectors.containsKey(catalogName), "Connector of catalog '%s' already exists", catalogName);
 
         Connector connector = connectorFactory.createConnector(context);
         connectors.put(catalogName, connector);
-        registerConnectorInternal(connector, context);
+
+        try {
+            registerConnectorInternal(connector, context);
+        } catch (Exception e) {
+            connectors.remove(catalogName);
+            throw new DdlException(String.format("Failed to create connector on [catalog : %s, type : %s]",
```

Comment:
this will cause fe start with failure if connector created for one of the catalog failed ? This behavior may need to be discussed

Reply:
I will test it when we have implemented persistent catalog.